provisioning/src/Azure/AzCli.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Provisioning\Azure;
+
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+
+final class AzCli
+{
+    private string $azureServicePrincipalTenant;
+
+    private string $azureServicePrincipal;
+
+    private string $azureServicePrincipalPassword;
+
+    public function __construct(
+        string $azureServicePrincipalTenant,
+        string $azureServicePrincipal,
+        string $azureServicePrincipalPassword
+    ) {
+        $this->azureServicePrincipalTenant = $azureServicePrincipalTenant;
+        $this->azureServicePrincipal = $azureServicePrincipal;
+        $this->azureServicePrincipalPassword = $azureServicePrincipalPassword;
+    }
+
+    public function runAuthorizedSingleLineOutputCmd(string $cmd): string
+    {
+        $out = $this->runAuthorizedCmd(
+            $cmd
+        );
+        return explode(PHP_EOL, $out)[0];
+    }
+
+    public function runAuthorizedCmd(string $cmd): string
+    {
+        $volume = __DIR__ . '/../../../';
+        // phpcs:disable
+        $cmd = <<< EOT
+    docker run --volume $volume:/keboola \
+        -e AZURE_SERVICE_PRINCIPAL_PASSWORD=$this->azureServicePrincipalPassword \
+        -e AZURE_SERVICE_PRINCIPAL=$this->azureServicePrincipal \
+        -e AZURE_SERVICE_PRINCIPAL_TENANT=$this->azureServicePrincipalTenant \
+        quay.io/keboola/azure-cli \
+        sh -c 'az login --service-principal -u \$AZURE_SERVICE_PRINCIPAL -p \$AZURE_SERVICE_PRINCIPAL_PASSWORD --tenant \$AZURE_SERVICE_PRINCIPAL_TENANT >> /dev/null && $cmd'
+EOT;
+        // phpcs:enable
+
+        $process = Process::fromShellCommandline($cmd);
```

Comment:
tady mi přijde škoda řešit escapování ručně, když právě process umí všechny ty věci oescapovat správně :(

Reply:
To mě přijde škoda to přepisovat na tu array syntaxy :D Udělat tam chybu v takovém cmd je strašně jednoduché a debilně sa to hledá takto to máš přímo před očima.